obspy/core/inventory/inventory.py

```diff
@@ -1117,6 +1117,27 @@ def plot_response(self, min_freq, output="VEL", network="*", station="*",
 
         return fig
 
+    def sort(self):
```

Comment:
How about something like this:

```suggestion
    def sort(self, keys=("code", "start_date"), max_level="channel"):
        sort_fct = lambda x: [getattr(x, i) for i in keys]
        self.networks = sorted(self.network, key=sort_fct)
        ...
```

Then it works for all keys and any number of keys (at least those that are present on all object types which could also checked before any in-place sorting takes place). I also added a `max_level` argument so one could for example only sort the networks and not the stations/channels/...

I also think the `.sort()` method should be part of every network/station/... object. If you implement it at the `BaseNode` (https://github.com/obspy/obspy/blob/master/obspy/core/inventory/util.py#L26) this would just work I think.